stan/math/opencl/kernel_generator/select.hpp

```diff
@@ -0,0 +1,126 @@
+#ifndef STAN_MATH_OPENCL_KERNEL_GENERATOR_SELECT_HPP
+#define STAN_MATH_OPENCL_KERNEL_GENERATOR_SELECT_HPP
+#ifdef STAN_OPENCL
+
+#include <stan/math/prim/meta.hpp>
+#include <stan/math/opencl/matrix_cl_view.hpp>
+#include <stan/math/opencl/kernel_generator/type_str.hpp>
+#include <stan/math/opencl/kernel_generator/name_generator.hpp>
+#include <stan/math/opencl/kernel_generator/operation_cl.hpp>
+#include <stan/math/opencl/kernel_generator/as_operation_cl.hpp>
+#include <stan/math/opencl/kernel_generator/is_valid_expression.hpp>
+#include <stan/math/opencl/kernel_generator/common_return_scalar.hpp>
+#include <set>
+#include <string>
+#include <type_traits>
+#include <utility>
+
+namespace stan {
+namespace math {
+
+/**
+ * Represents a selection operation in kernel generator expressions.
```

Comment:
[bikeshedding]

This is a little too broad, why not just say "Represents an ternary matrix operation similar to Eigen's select statement"

Reply:
Or like what you have below

>  Selection operation on kernel generator expressions. This is element wise ternary operator <code> condition ? then : els </code>.

Reply:
done